modules/data.land/R/extract_soil_nc.R

```diff
@@ -0,0 +1,195 @@
+#' Extract soil data
+#'
+#' @param in.file 
+#' @param outdir 
+#' @param lat 
+#' @param lon 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' in.file <- "~/paleon/env_paleon/soil/paleon_soil.nc"
+#' outdir  <- "~/paleon/envTest"
+#' lat     <- 40
+#' lon     <- -80
+#' \donotrun{
+#'    PEcAn.data.land::extract_soil_nc(in.file,outdir,lat,lon)
+#' }
+extract_soil_nc <- function(in.file,outdir,lat,lon){
+  
+  ## open soils
+  nc <- ncdf4::nc_open(in.file)
+  
+  ## extract lat/lon
+  dims <- names(nc$dim)
+  lat.dim <- dims[grep("^lat",dims)]
+  lon.dim <- dims[grep("^lon",dims)]
+  soil.lat <- ncdf4::ncvar_get(nc, lat.dim)
+  soil.lon <- ncdf4::ncvar_get(nc, lon.dim)
```

Comment:
Exactly, the names can differ (some files are just lat/lon rather than writing out latitude and longitude), hence the regexp above which ensures that we fix the names when we produce our file.